pkg/api/testapi/testapi.go

```diff
@@ -24,63 +24,144 @@ import (
+// GroupAndVersion returns the API version to test against for a group, as set
+// by the KUBE_TEST_API env var.
+// Return value is in the form of "group/version".
+func (g TestGroup) GroupAndVersion() string {
+	return g.GroupVersionUnderTest
 }
 
 // Codec returns the codec for the API version to test against, as set by the
-// KUBE_API_VERSION env var.
-func Codec() runtime.Codec {
-	interfaces, err := latest.InterfacesFor(Version())
-	if err != nil {
-		panic(err)
+// KUBE_TEST_API env var.
+func (g TestGroup) Codec() runtime.Codec {
+	if g.Group == "" {
+		interfaces, err := latest.InterfacesFor(g.GroupVersionUnderTest)
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.Codec
 	}
-	return interfaces.Codec
+	if g.Group == "experimental" {
+		interfaces, err := explatest.InterfacesFor(g.GroupVersionUnderTest)
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.Codec
+	}
+	panic(fmt.Errorf("cannot test group %s", g.Group))
 }
 
 // Converter returns the api.Scheme for the API version to test against, as set by the
-// KUBE_API_VERSION env var.
-func Converter() runtime.ObjectConvertor {
-	interfaces, err := latest.InterfacesFor(Version())
-	if err != nil {
-		panic(err)
+// KUBE_TEST_API env var.
+func (g TestGroup) Converter() runtime.ObjectConvertor {
+	if g.Group == "" {
+		interfaces, err := latest.InterfacesFor(g.GroupVersionUnderTest)
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.ObjectConvertor
+	}
+	if g.Group == "experimental" {
+		interfaces, err := explatest.InterfacesFor(g.GroupVersionUnderTest)
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.ObjectConvertor
 	}
-	return interfaces.ObjectConvertor
+	panic(fmt.Errorf("cannot test group %s", g.Group))
+
 }
 
 // MetadataAccessor returns the MetadataAccessor for the API version to test against,
-// as set by the KUBE_API_VERSION env var.
-func MetadataAccessor() meta.MetadataAccessor {
-	interfaces, err := latest.InterfacesFor(Version())
-	if err != nil {
-		panic(err)
+// as set by the KUBE_TEST_API env var.
+func (g TestGroup) MetadataAccessor() meta.MetadataAccessor {
+	if g.Group == "" {
+		interfaces, err := latest.InterfacesFor(g.GroupVersionUnderTest)
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.MetadataAccessor
+	}
+	if g.Group == "experimental" {
+		interfaces, err := explatest.InterfacesFor(g.GroupVersionUnderTest)
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.MetadataAccessor
 	}
-	return interfaces.MetadataAccessor
+	panic(fmt.Errorf("cannot test group %s", g.Group))
 }
 
 // SelfLink returns a self link that will appear to be for the version Version().
 // 'resource' should be the resource path, e.g. "pods" for the Pod type. 'name' should be
 // empty for lists.
-func SelfLink(resource, name string) string {
+func (g TestGroup) SelfLink(resource, name string) string {
 	if name == "" {
-		return fmt.Sprintf("/api/%s/%s", Version(), resource)
+		return fmt.Sprintf("/api/%s/%s", g.Version(), resource)
```

Comment:
Hm, shouldn't the group appear in this somewhere? We mount experimental under a different path, right?